data/kernels/guided_filter.cl

```diff
@@ -17,7 +17,7 @@
 */
 
 #include "common.h"
-
+#pragma OPENCL EXTENSION cl_khr_fp64 : enable
```

Comment:
Is that mandatory ? I understand that a pragma is just a request to the compiler that may or may not be honored, but what happens if there is no such OpenCL extension available ?

Reply:
This is mandatory to enable double precision floating point arithmetic. I revised my code to use the [Kahan summation algorithm](https://en.wikipedia.org/wiki/Kahan_summation_algorithm), which allows me to implement the sliding mean entirely in single precision without losing (too much of) numerical precision. In this way I was able to get rid of the pragma avoiding possible portability issues.